app/controllers/posts_controller.rb

```diff
@@ -8,5 +9,24 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
-  # 1
+
+  def create
+    new_post = current_user.posts.new(post_params)
+    new_post.likes_counter = 0
+    new_post.comments_counter = 0
+    new_post.update_posts_counter
+    respond_to do |format|
+      format.html do
+        if new_post.save
+          redirect_to "/users/#{new_post.user.id}/posts/", notice: 'Success!'
+        else
+          render :new, alert: 'Error!'
+        end
+      end
+    end
+  end
+
+  def post_params
+    params.require(:data).permit(:title, :text)
+  end
```

Comment:
- [ ] I would recommend making `post_params` the private method, it makes the application secure by permitting only limited attributes.